src/Storage/FileStorage.php

```diff
@@ -1,476 +1,365 @@
+            return (string)$data;
+        }
+
+        $createdAt = $this->getDateHelper()->getCurrentTimestamp();
+        $expiresAt = 0;
+
+        if ($ttlInSeconds > 0) {
+            $expiresAt = $createdAt + $ttlInSeconds;
+        }
+
+        $file = new File($data, $createdAt, $expiresAt);
+
+        return json_encode($file);
+    }
+
+    /**
+     * Processes the data read from the file
+     *
+     * @throws StorageException
+     */
+    protected function readData(string $fileContent): File
+    {
+        if ($this->canOnlyStorePlainText) {
+            return new File($fileContent);
+        }
+
+        /** @var File $fileContent */
+        $encodedFileContent = json_decode($fileContent);
+
+        if (!property_exists($encodedFileContent, 'data') || !property_exists($encodedFileContent, 'expiresAt')) {
+            throw new StorageException('Unable to retrieve stored data');
+        }
+
+        return $encodedFileContent;
+    }
+
+    protected function isExpired(File $file): bool
+    {
+        $currentTime = $this->getDateHelper()->getCurrentTimestamp();
+
+        if (!empty($file->expiresAt) && $file->expiresAt < $currentTime) {
+            return true;
+        }
+
+        return false;
+    }
+
+    /**
+     * Deletes the data specified by the key
+     *
+     * @throws StorageException
+     */
+    public function delete(string $key): void
+    {
+        $item = (new Storage(Storage::METHOD_DELETE, $key));
+
+        $this->protectWhenReadOnly();
+
+        $fileName = $this->getFullPath($key);
+
+        try {
+            $this->fileHandler->remove($fileName);
+        } catch (NotFoundException $e) {
+        } catch (FileException $e) {
+            throw new StorageException('Unable to remove the file: ' . $fileName, 0, $e);
+        }
+
+        $item->setFinished();
+        $this->getDebugDataHandlerRegistry()->addStorage($item);
+    }
+
+    /**
+     * Deletes every data in the storage.
+     *
+     * @throws StorageException
+     */
+    public function clear(): void
+    {
+        $item = (new Storage(Storage::METHOD_CLEAR));
+
+        $this->protectWhenReadOnly();
+
+        try {
+            $this->fileHandler->removeDirectory($this->path, true);
+        } catch (FileException $e) {
+            throw new StorageException('Unable to remove the directory: ' . $this->path, 0, $e);
+        }
+
+        $item->setFinished();
+        $this->getDebugDataHandlerRegistry()->addStorage($item);
+    }
+
+    public function isPersistent(): bool
+    {
+        // File storage is always persistent.
+        return true;
+    }
+
+    public function isTtlSupported(): bool
+    {
+        // If the storePlainText option is set to false, we support TTL functionality.
+        return !$this->canOnlyStorePlainText;
+    }
+
+    public function isReadOnly(): bool
+    {
+        return $this->readOnly;
+    }
+
+    public function isCanOnlyStorePlainText(): bool
```

Comment:
The is prefix is completely unnecessary, canOnlyStorePlainText() implies it's a boolean as well

Reply:
PHPstorm